aten/src/ATen/native/sparse/SparseMatMul.cpp

```diff
@@ -230,16 +230,20 @@ void sparse_matmul_kernel(
   Tensor output_row_indices = output_indices.select(0, 0);
   Tensor output_col_indices = output_indices.select(0, 1);
 
+  // Get contiguous mat1 and mat2 values, since they will be used via data_ptr.
+  auto mat1_csr_values = mat1_csr.values().contiguous();
+  auto mat2_csr_values = mat2_csr.values().contiguous();
+
   // TODO: replace with a CSR @ CSC kernel for better performance.
   _csr_matmult(
       M,
       N,
       mat1_csr.crow_indices().data_ptr<int64_t>(),
       mat1_csr.col_indices().data_ptr<int64_t>(),
-      mat1_csr.values().data_ptr<scalar_t>(),
+      mat1_csr_values.data_ptr<scalar_t>(),
```

Comment:
nit: `_csr_matmul` is a custom kernel, right? Is it easy to make it stride-aware instead? Otherwise it could be done in a follow-up?

Reply:
The kernel uses the Strided accessor as a wrapper. The last argument is a stride. We could replace 1 with the corresponding stride value. Let's also add a test, shall we? Similar technique could be applied for index values as well.